src/lib.rs

```diff
@@ -0,0 +1,198 @@
+        if 0 == path_nodes.len() {
+            panic!("Need a least one node for registering a value...");
+        }
+        return self.route.register(path_nodes, value);
+    }
+
+    pub fn find(&self, path_nodes : &[PathNode]) -> Result<RouteValue, SisoError> {
+        if 0 == path_nodes.len() {
+            panic!("Need a least one node for finding a value...");
+        }
+        return self.route.find(path_nodes);
+    }
+
+    pub fn new() -> Siso {
+        Siso { route: Route::new(PathNode::PathPart(String::from("[root]")), None) }
+    }
+}
+
+#[derive(Debug)]
+struct Route {
+    node: PathNode,
+    routes: Vec<Route>,
+    value: RouteValue,
+}
+
+impl Route {
+    fn register(&mut self, path_nodes : &[PathNode], value: RouteValue) -> Result<bool, SisoError> {
+        let path_node = &path_nodes[0];
+        let mut index = Err("");
+
+        for i in 0..self.routes.len() {
+            if *path_node == self.routes[i].node {
+                index = Ok(i)
+            }
+        }
+
+        let index = match index {
+            Ok(i) => i,
+            _ => {
+                let new_route = Route::new(path_node.clone(), None);
+                self.routes.push(new_route);
+                self.routes.len() - 1
+            }
+
+        };
+
+        if path_nodes.len() == 1 {
+            return match &self.routes[index].value {
+                RouteValue::Value(_) => Err(SisoError { code: "E_VALUE_EXISTS".to_string() }),
+                RouteValue::None => {
+                    self.routes[index].value = value;
+                    Ok(true)
+                },
+            }
+        } else {
+            return self.routes[index].register(&path_nodes[1..], value);
+        }
+    }
+    fn find(&self, path_nodes : &[PathNode]) -> Result<RouteValue, SisoError> {
+        for a_route in self.routes.iter() {
+            if path_nodes[0] == a_route.node {
+                if path_nodes.len() == 1 {
+                    return Ok(a_route.value.clone());
+                } else {
+                    return a_route.find(&path_nodes[1..]);
+                }
+            }
+        }
+        Err(SisoError { code: "E_NOT_FOUND".to_string() })
+    }
+
+    fn new(node : PathNode, value : Option<RouteValue> ) -> Route {
+        Route { routes: Vec::new(), node, value: value.unwrap_or(RouteValue::None) }
+    }
+}
+
+// TODO: Allow generic value instead of strings only
+#[derive(Debug, Clone)]
+pub enum RouteValue {
+    Value(String),
+    None,
+
+}
+
+impl PartialEq for RouteValue {
+    fn eq(&self, other: &RouteValue) -> bool {
+        match self {
+            RouteValue::Value(self_value) => {
+                match other {
+                    RouteValue::Value(other_value) => self_value == other_value,
+                    RouteValue::None => false,
+                }
+            },
+            RouteValue::None => false,
```

Comment:
Is this intentional that `RouteValue::None != RouteValue::None`? If it is, I think a comment would help!

Reply:
Yes, it means "no value for the path x" a bit like NaN != NaN but indeed, I'll add a comment here to explain this.